paimon-core/src/main/java/org/apache/paimon/disk/SpillableBuffer.java

```diff
@@ -53,14 +53,18 @@ public class ExternalBuffer {
 
     // The size of each segment
     private final int segmentSize;
+    private final boolean spillable;
 
     private final List<ChannelWithMeta> spilledChannelIDs;
     private int numRows;
 
     private boolean addCompleted;
 
-    public ExternalBuffer(
-            IOManager ioManager, MemorySegmentPool pool, AbstractRowDataSerializer<?> serializer) {
+    public SpillableBuffer(
+            IOManager ioManager,
+            MemorySegmentPool pool,
+            AbstractRowDataSerializer<?> serializer,
+            boolean spillable) {
```

Comment:
You should first introduce a new interface, and ExternalBuffer and InMemoryBuffer implements it.

Reply:
done